### PR TITLE
CI: update to ruby 2.4.1 and gem update

### PR DESCRIPTION
**Pull request by artem-sidorenko**
Chef 13 is also using ruby 2.4.1 in the omnibus packages

There are problems with Chef 13.1 in the chefspec tests. The builds are getting stucked without any specific reason. I'll investigate this further...

### CODE REVIEW

**Comment by artem-sidorenko** [comment]
@chris-rock @atomic111 can somebody please have a look here? I would like to make a new release when this is merged

File: recipes/pam.rb [hunk 1]
```diff
@@ -71,7 +72,7 @@
   end
 
   # configure tally2
-  if node['os-hardening']['auth']['retries'] > 0
+  if node['os-hardening']['auth']['retries'].positive?
```
Comment (atomic111) [inline recipes/pam.rb]: do you know, why there is the error with the ubuntu 16.04 machine with chef 12.5

Comment (artem-sidorenko) [inline recipes/pam.rb]: funny thing, it looks like chef 12.5.1 was packaged with different ruby for ubuntu then to centos. I'll have a detailed look tomorrow

Comment (artem-sidorenko) [inline recipes/pam.rb]: @atomic111 fixed. chef 12.5.1 has ruby 2.1 and not 2.3 (2.1 does not have `positive?`), I changed the rubocop settings to reflect that and also created https://github.com/dev-sec/chef-ssh-hardening/pull/179

Comment (artem-sidorenko) [inline recipes/pam.rb]: @atomic111 thanks for catching this btw :-)

**Comment by artem-sidorenko** [comment]
something is broken with NYC2 DO region: its temporary disabled by DO completely. If it is not getting fixed today I'll switch our tests to the NYC3 region

**Comment by artem-sidorenko** [comment]
@atomic111 fixed now (I switched the tests to the NYC3 region)

**Pull request merged**